modules/calib3d/src/usac/fundamental_solver.cpp

```diff
@@ -0,0 +1,337 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "../precomp.hpp"
+#include "../usac.hpp"
+#ifdef HAVE_EIGEN
+#include <Eigen/Eigen>
+#endif
+
+namespace cv { namespace usac {
+// Fundamental Matrix Solver:
+class FundamentalMinimalSolver7ptsImpl: public FundamentalMinimalSolver7pts {
+private:
+    const Mat * points_mat;
+    const float * const points;
+public:
+    explicit FundamentalMinimalSolver7ptsImpl (const Mat &points_) :
+            points_mat (&points_), points ((float *) points_.data) {}
+
+    int estimate (const std::vector<int>& sample, std::vector<Mat> &models) const override {
+        const int m = 7, n = 9; // rows, cols
+        std::vector<double> a(m*n);
+
+        int cnt = 0;
+        for (int i = 0; i < m; i++ ) {
+            const int smpl = 4*sample[i];
+            const auto x1 = points[smpl  ], y1 = points[smpl+1],
+                       x2 = points[smpl+2], y2 = points[smpl+3];
+
+            a[cnt++] = x2*x1;
+            a[cnt++] = x2*y1;
+            a[cnt++] = x2;
+            a[cnt++] = y2*x1;
+            a[cnt++] = y2*y1;
+            a[cnt++] = y2;
+            a[cnt++] = x1;
+            a[cnt++] = y1;
+            a[cnt++] = 1;
+        }
+
+        Math::eliminateUpperTriangluar(a, m, n);
```

Comment:
Small typo :)

Reply:
Sorry, I don't really see a typo. Where is it? However, thanks for point it out, I will get rid of cnt++ and use pointer++ instead.

Reply:
Triangluar -> Triangular

Reply:
Oh, I see, thanks!